lib/workers/repository/updates/generate.ts

```diff
@@ -166,28 +167,30 @@ export function generateBranchConfig(
     } else if (semver.valid(toVersions[0])) {
       upgrade.isRange = false;
     }
+    const commitMessageFactory = new CommitMessageFactory(upgrade);
+    const commitMessage = commitMessageFactory.create();
     // Use templates to generate strings
-    if (upgrade.semanticCommits === 'enabled' && !upgrade.commitMessagePrefix) {
+    if (SemanticCommitMessage.is(commitMessage)) {
       logger.trace('Upgrade has semantic commits enabled');
-      let semanticPrefix = upgrade.semanticCommitType;
-      if (upgrade.semanticCommitScope) {
-        semanticPrefix += `(${template.compile(
-          upgrade.semanticCommitScope,
-          upgrade
-        )})`;
-      }
-      upgrade.commitMessagePrefix = CommitMessage.formatPrefix(semanticPrefix);
-      upgrade.toLowerCase =
-        regEx(/[A-Z]/).exec(upgrade.semanticCommitType) === null &&
-        !upgrade.semanticCommitType.startsWith(':');
+      commitMessage.scope = template.compile(
+        upgrade.semanticCommitScope,
+        upgrade
+      );
     }
     // Compile a few times in case there are nested templates
-    upgrade.commitMessage = template.compile(
+    commitMessage.subject = template.compile(
       upgrade.commitMessage || '',
       upgrade
     );
-    upgrade.commitMessage = template.compile(upgrade.commitMessage, upgrade);
-    upgrade.commitMessage = template.compile(upgrade.commitMessage, upgrade);
+    commitMessage.subject = template.compile(
+      commitMessage.toJSON().subject,
```

Comment:
```suggestion
      commitMessage.subject,
```
i don't like this, can we make the getter public instead

Reply:
Done